### DISCUSSION TITLE
Horizontal scroll beyond end of longest line in visible text area ?

### DISCUSSION THREAD

**Discussion by grege2** [eteran/nedit-ng#305] [General]
Hi, this may turn out a question which there's already an obvious answer to. Which I'll be happy to find out !

Quite often I find myself wanting to scroll the view beyond the end of the longest line in the visible text area. Mostly it's code, or in fact data being set up in code ...

```
a = 1;
data[1] =  "FFEE1234 .................................................. maybe 200 chars of stuff ................................................AABB1122";
```

NG will only scroll horizontally until the end of the line is against the RHS of the text window. Very logical, but sometimes it would be useful to scroll further. Example ..

```
data[1] = "FFEE1234 .................................................. maybe 200 chars of stuff ..............................................................AABB1122";
data[2] = "FFEE1234 .................................................. maybe 200 chars of stuff ...........important small differerences in line endings here AABB1122 X";
data[3] = "FFEE1234 .................................................. maybe 200 chars of stuff ...........important small differerences in line endings here AABB1122 YYY";
data[4] = "FFEE1234 .................................................. maybe 200 chars of stuff ...........important small differerences in line endings here AABB1122 ZZ";
data[5] = "FFEE1234 .................................................. maybe 200 chars of stuff ...........important small differerences in line endings here AABB1122 WWWW";
... etc
```
So, I would like to a) scroll further to the right than NG currently allows, so there's clear space at the end of the lines, to see things better b) I'd like NG to be able to stay "fixed" at it's current scroll pos as I scroll the file up and down. NG currently "snaps" back to align at the right of the longest line in the visible buffer. 

As I write this I'm thinking "ok .. I could have a macro that pads each line with spaces, to a length of say longest-line + 50 chars, then the horiz scrolling would stay put, and line endings would be clearly viewable". Clunky but in fact ok for my use (Javascript, trailing spaces have no effect, and I already strip them routinely before compile). But it would be cool if I could "lock" the horiz scrolling where I want it. 

I have no idea whether other editors provide this - the fact that it often frustrates me may be a dim memory of Emacs, so long ago I don't want to tell you, or a thing called SGI "jot" or "zip", which were quite nedit-like.

Thanks for any thoughts.

**Comment by grege2** [comment]
Sorry, it's better to view this item on the GitHib Discussions page, the scrolling of the fixed-width code box illustrates what I'm talking about, the email version kind of scrambles it.

**Comment by eteran** [comment]
This is a bit of an "unusual" request, but I'll put some thought into what the best way to accommodate this would be :-) Since I'm not sure if there are any editors that handle horizontal scrolling quite this way.

**Comment by grege2** [comment]
Hey Evan, thanks. I'll do some research too. And maybe get my "pad lines to fixed length" macro going. I don't know if other editors provide this, it seems simple to be able to scroll beyond the end the line, and yet .. not often needed maybe. Definitely a "6.2+" feature.

**Comment by grege2** [comment]
Ok, here's a macro which pads the selection, assumed to be nn complete lines, or maybe the whole file, to a fixed length with spaces, or some other character. Then you can scroll to the right and see line endings more easily as outlined above. If you pad with spaces, in many languages you can just leave them there. Or strip them sometime at your leisure. Of course the file may be data or plain text or anything .. but the same remarks generally apply.

So this 3/4 solves my issue, so the request for the built-in NG scrolling enhancement can go on some 6.x todo list.  Ciao !

```
define pad_selection_to_fixed_length {
  
  pad_length = string_dialog("pad length", "ok")                          # perhaps 150 or 200 
  pad_char = string_dialog("pad character", "ok")                         # typically a space char
  end_marker = string_dialog("end marker (hit return for none)", "ok")    # optional end marker
  
  end_of_selection() 
  last_position = $cursor      # remember end of selection
  
  beginning_of_selection()     
  
  while($cursor < last_position) {
    end_of_line()
    end_col = $column
    extra_spaces = pad_length - end_col
    extra_string = ""
    for(j=0; j<extra_spaces; j++)
      extra_string = extra_string pad_char   # Inefficient, but I don't see a better way. Eg. JS let padding = padstr.repeat(n);
    insert_string(extra_string)
    insert_string(end_marker)  # optional end marker 
    
    beginning_of_line()
    process_down()  
  }
}
```
